File: ZYSwiftUIFrame/ZYSwiftUIFrame/UI/Message/Message.swift
//
//  Message.swift
//  ZYSwiftUIFrame
//
//  Created by lzy on 2022/3/31.
//

/**
 * 消息实体类
 */
struct Message: Codable, Identifiable, Equatable {
    static func == (lhs: Message, rhs: Message) -> Bool {
        return lhs.content == rhs.content
        && lhs.theme == rhs.theme
    }
    
    @Default<Int> var id: Int
    @Default<String> var theme: String // 标题
    @Default<String> var content: String // 内容
    @Default<Int> var receiveUser: Int // 接收人id
    @Default<Int> var sendUser: Int // 发送人id
    @Default<Int> var isRead: Int // 已读 0否1是
    @Default<String> var createdAt: String // 创建时间
    
    init() {
        self.id = -1
        self.receiveUser = -1
        self.sendUser = -1
        self.theme = ""
        self.content = ""
        self.isRead = 0
        self.createdAt = "暂时没有时间数据......"
    }
    
}

File: go_api_server/model/message.go
package model

type Message struct {
	BaseModel
	Theme       string `json:"theme" gorm:"comment:消息标题"`
	Content     string `json:"content" gorm:"comment:消息内容"`
	IsRead      int    `json:"isRead" gorm:"comment:是否已读"`
	ReceiveUser int    `json:"receiveUser" gorm:"comment:接收人id"`
	SendUser    int    `json:"sendUser" gorm:"comment:发送人id"`
}
